src/hooks/useWithdrawLiquidityHandler.tsx

```diff
@@ -22,342 +18,201 @@ export const useWithdrawLiquidityHandler = () => {
+      })
   }
 
   async function withdrawLiquidity(balance: number) {
+    if (balance === 0) {
+      throw new Error('Please select a valid amount to withdraw')
+    }
     let amt = '0'
     if (balance === 100) {
       amt = totalLiquidityAmt.toString()
     } else {
       amt = ethers.utils.parseEther(((liquidityAmt * balance) / 100).toString()).toString()
     }
-    // Position Manager contract address
-    let address =
-      process.env.NEXT_PUBLIC_ENVIRONMENT === 'local'
-        ? process.env.NEXT_PUBLIC_POSITION_MANAGER_ADDRESS || ''
-        : DEPOSIT_POOL_ADDRESS
-    if (provider) {
-      if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-        if (accountInfo && accountInfo?.address) {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider.getSigner(accountInfo?.address)))
-        } else {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider))
-        }
-      } else {
-        if (accountInfo && accountInfo?.address) {
-          setDepPool(new ethers.Contract(address, DepPool.abi, provider.getSigner(accountInfo?.address)))
-        } else {
-          setDepPool(new ethers.Contract(address, DepPool.abi, provider))
-        }
-      }
-    } else {
-      notifyError('Please connect wallet')
-    }
-
     if (!accountInfo || !accountInfo.address) {
       notifyError('Wallet not connected.')
       return
     }
 
-    if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-      if (posManager === null) {
-        return
-      }
+    if (parseInt(amt) === 0) {
+      throw new Error('User does not have enough reserves to withdraw')
+    }
 
-      try {
-        const WithdrawReservesParams = {
-          cfmm: process.env.NEXT_PUBLIC_CFMM_ADDRESS,
-          protocol: 1,
-          amount: amt,
-          amountsMin: [100, 200, 300],
-          to: accountInfo.address,
-          deadline: ethers.constants.MaxUint256,
-        }
-        let tx = await posManager.withdrawReserves(WithdrawReservesParams)
-        return await tx.wait()
-      } catch (e) {
-        return e
+    try {
+      if (!posManager) {
+        await initializePositionManagerContract()
       }
-    } else {
-      if (depPool === null) {
-        return
+
+      const WithdrawReservesParams = {
+        cfmm: process.env.NEXT_PUBLIC_CFMM_ADDRESS,
+        protocol: 1,
+        amount: amt,
+        amountsMin: [100, 200, 300],
+        to: accountInfo.address,
+        deadline: ethers.constants.MaxUint256,
       }
 
-      try {
-        let tx = await depPool.removeLiquidity(amt, ZEROMIN, ZEROMIN, accountInfo.address, {
-          gasLimit: process.env.NEXT_PUBLIC_GAS_LIMIT,
-        })
+      if (posManager) {
+        let tx = await posManager.withdrawReserves(WithdrawReservesParams)
         return await tx.wait()
-      } catch (e) {
-        return e
       }
+    } catch (e) {
+      throw new Error('An error occurred while whithdrawing reserves')
```

Comment:
Would it be better to use notifyError?

Reply:
Agreed

Reply:
All the throw statements are going to be shown in notifyError in the main method